flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/deduplicate/MiniBatchDeduplicateKeepFirstRowFunction.java

```diff
@@ -42,18 +44,25 @@
 	private static final long serialVersionUID = -7994602893547654994L;
 
 	private final TypeSerializer<BaseRow> typeSerializer;
-
+	private final long minRetentionTime;
+	private final boolean stateCleaningEnabled;
 	// state stores a boolean flag to indicate whether key appears before.
 	private ValueState<Boolean> state;
 
-	public MiniBatchDeduplicateKeepFirstRowFunction(TypeSerializer<BaseRow> typeSerializer) {
+	public MiniBatchDeduplicateKeepFirstRowFunction(TypeSerializer<BaseRow> typeSerializer, long minRetentionTime) {
+		this.minRetentionTime = minRetentionTime;
+		stateCleaningEnabled = minRetentionTime > 1;
 		this.typeSerializer = typeSerializer;
 	}
 
 	@Override
 	public void open(ExecutionContext ctx) throws Exception {
 		super.open(ctx);
 		ValueStateDescriptor<Boolean> stateDesc = new ValueStateDescriptor<>("existsState", Types.BOOLEAN);
+		StateTtlConfig ttlConfig = createTtlConfig(minRetentionTime, stateCleaningEnabled);
+		if (ttlConfig.isEnabled()){
```

Comment:
`if (ttlConfig.isEnabled()) {`
we always add a space before `{`

Reply:
got it